### PR TITLE
Add fundRewards to sc

### PR DESCRIPTION
**Pull request by nz-neooo**

### CODE REVIEW

File: contracts/StateChannel.sol [hunk 1]
```diff
@@ -254,8 +257,33 @@ contract StateChannel is Initializable, OwnableUpgradeable {
...
+
+        address consumer = channels[channelId].consumer;
+        if (msg.sender != consumer) {
+            bool isController = IConsumerRegistry(settings.getContractAddress(SQContracts.ConsumerRegistry)).isController(consumer, msg.sender);
+            require(isController, 'SC012');
+        }
+
+        // TODO: transfer the balance to contract
+        address rbAddress = settings.getContractAddress(SQContracts.RewardsBooster);
+        IRewardsBooster(rbAddress).spent(consumer, amount);
```
Comment (nz-neooo) [inline contracts/StateChannel.sol]: Do we need to limit the `charge amount <= realTotal` ?

**Comment by ianhe8x** [comment]
@nz-neooo summary
* support refund unused query rewards back to rewardsbooster
* rewardsbooster.spend() returns the available amount consumer can spend
* merge  fundRewards() with original fund(), use query rewards as much as possible

File: contracts/RewardsBooster.sol [hunk 2]
```diff
@@ -543,4 +543,13 @@ contract RewardsBooster is Initializable, OwnableUpgradeable, IRewardsBooster {
     function getBoosterQueryRewards(bytes32 _deploymentId, address _account) view external returns (BoosterQueryReward memory)  {
         return deploymentPools[_deploymentId].boosterQueryRewards[_account];
     }
+
+    function spent(address _indexer, uint256 amount) external {
```
Comment (ianhe8x) [inline contracts/RewardsBooster.sol]:
```suggestion
    function spendQueryRewards(address _indexer, uint256 amount) external returns(uint256) {
```

File: contracts/StateChannel.sol (hunk 1 above)
Comment (ianhe8x) [reply contracts/StateChannel.sol]: That's what we debated right? if consumer can not withdraw the over-funded token, and we can refund them. 
I  don't mind if amount > realTotal

**Comment by nz-neooo** [comment]
Need fix spendQueryRewards in booster, fund/refund state channel is ok. @ianhe8x

**Review by ianhe8x — APPROVED** [review]

**Pull request merged**